libnmstate/nm/connection.py

```diff
@@ -68,9 +68,9 @@ def update_profile(base_profile, new_profile):
     base_profile.replace_settings_from_connection(new_profile)
 
 
-def commit_profile(connection_profile, save_to_disk=True):
+def commit_profile(connection_profile, save_to_disk=True, metadata=None):
```

Comment:
AFAICS, metadata should be always called nmdev since the callback assumes that it will be nmdev and not arbitrary metadata.